implemented revealing object names

contriteobserver opened the pull request:
by prefixing each generated child with the name of the source material that caused the object to be created.

addresses #2301

contriteobserver committed:
implemented revealing object names, by prefixing each generated child with the name of the
source material that caused the object to be created.

hannesa2 commented on rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java:
```diff
@@ -416,7 +416,12 @@ private void mergeGroupsAsObjects(Object3D object) {
+
+	private static String generateObjectName(String prefix) {
+		if(prefix==null) prefix = "Object";
+		return prefix + (int) (Math.random() * 9e3 + 1e3);
```
Why not using a timestamp (or UUID) ?

contriteobserver replied:
Excellent idea, I went with a timestamp as the ramifications of that system call is better understood.

ToxicBakery replied:
Seeing this now after my other question. I really think uuid implies the intent vs a random number or nanos usage.

contriteobserver committed:
converted uniqueness portion of generated name to
a string representation  of `SystemClock.elapsedRealtime()`

hannesa2 commented on rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java:
```diff
@@ -421,7 +422,7 @@ private static String generateObjectName() {
 	private static String generateObjectName(String prefix) {
 		if(prefix==null) prefix = "Object";
-		return prefix + (int) (Math.random() * 9e3 + 1e3);
+		return prefix + Long.toString(SystemClock.elapsedRealtime());
```
Keep in mind, this is in nanos, I would go with millis `System.currentTimeMillis()` which is more common to use as unixtimestamp. But I'm fine with it.

contriteobserver replied:
`System.currentTimeMillis()` would not necessarily provide the necessary resolution, as models sometimes duplicate names, and we can parse multiple child objects in a mS

object graph
    -->GROUP default
    	-->Arm.Left2597618548
    	-->Arm.Right2597618769
    	-->Leg.Left2597619008
    	-->Leg.Right2597619185
    	-->Head2597619359
    	-->Standing_Pose

jonathanchh1 replied:
I see you guys have now implemented this feature. Is there away to pull these changes or can it be released? I'm working on something really urgent and i would love to have this feature. Thank you!

hannesa2 replied:
Really urgent ? Nothing to do, here you go 
`implementation "com.github.contriteobserver:Rajawali:LoadedMtlObjNaming-SNAPSHOT"`
or
`implementation "com.github.contriteobserver:Rajawali:45cf99dc15"`

With jitpack.io you can have a library of any commit. Please see [details](https://jitpack.io/#contriteobserver/Rajawali/45cf99dc15)

jonathanchh1 replied:
Yes, definitely urgent! thank you again!

ToxicBakery replied:
If the object names need guaranteed uniqueness, why not a uuid?

contriteobserver replied:
UUIDs alone are insufficient, as there is a need for name to be revealing. Also, legacy identifiers had the classifier "Object" and the instance label "[0-9][0-9][0-9][0-9]" select-able by character type (letters vs. numbers) it seems appropriate to preserve that ability. 

Also, the overhead of a UUID call is concerning, as it must at least interface with the MAC address, the clock, pseudo-random generator, and a hash system. Was worried a UUID call might slow down file parsing.

Open to ideas if anyone has a suggestion.

ToxicBakery replied:
Maybe we should consider using an atomic number generator then. Maybe useful as the object IDs should be consistent as long as the model or loading order does not change.

Ie. AtomicInteger getAndIncrement()
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicInteger.html#getAndIncrement--

contriteobserver replied:
Implemented an `AtomicInteger` based identification mechanism, the `PosingDoll` object graph now looks like:

 I/MaterialNamingRenderer.InitScreen: object graph
    -->GROUP default
    	-->Arm.Left1002
    	-->Arm.Right1003
    	-->Leg.Left1004
    	-->Leg.Right1005
    	-->Head1006
    	-->Standing_Pose

contriteobserver committed:
now identifying unique Object3D instances via an AtomicInteger mechanism.

 I/MaterialNamingRenderer.InitScreen: object graph
    -->GROUP default
        -->Arm.Left1002
        -->Arm.Right1003
        -->Leg.Left1004
        -->Leg.Right1005
        -->Head1006
        -->Standing_Pose

addresses issue raised in pull request #2302 review
